File: api/Data/Appointment.cs
using Org.BouncyCastle.Asn1.Esf;

namespace api.Data
{
    public class Appointment
    {
        public Guid ID { get; set; }
        public string Staff_id { get; set; }
        public string Student_id { get; set; }
        public string DateTime { get; set; }

        public bool Approve { get; set; }
        public bool Pending { get; set; }
        public bool Complete { get; set; }
        public bool Cancel { get; set; }
        public string? Cancel_reason { get; set; }
        public string? Note { get; set; }
        public string? Staff_email { get; set; }
        public string? Student_email { get; set; }


        public Appointment(Guid id, string staff_id, string student_id, string datetime, 
                            bool approve, bool pending, bool complete, bool cancel, 
                            string cancel_reason, string note, string? staff_email, string? student_email)
        {
            ID = id;
            Staff_id = staff_id;
            Student_id = student_id;
            DateTime = datetime;
            Approve = approve;
            Pending = pending;
            Complete = complete;
            Cancel = cancel;
            Cancel_reason = cancel_reason;
            Note = note;
            Staff_email = staff_email;
            Student_email = student_email;
        }
        public Appointment()
        {
            ID = Guid.Empty;
            Staff_id = "";
            Student_id = "";
            DateTime = "";
            Approve = false;
            Pending = false;
            Complete = false;
            Cancel = false;
            Cancel_reason = "";
            Note = "";
            Staff_email = "";
            Student_email = "";
        }
    }
}

File: my-hd-app/src/page/Staff/Appointment.js
import React, { useState, useEffect, useContext } from 'react';
import { Container, Card, ListGroup, Breadcrumb, Button } from 'react-bootstrap'
import { useParams } from 'react-router-dom';
import axios from "axios"
import { AuthContext } from "../../context/auth";
import StatusMsg from "../../component/StatusMsg"

const Appointment = () => {
    const { staff } = useContext(AuthContext);
    const { id } = useParams();
    const [consultation, setConsultation] = useState(
        {
            ref_num: '',
            student: '',
            status: '',
            note: '',
            cancel_reason: '',
            datetime: ''
        }
    )

    useEffect(() => {
        axios({

            // Endpoint to send files
            url: "https://localhost:7261/api/v1/Appointment/" + id,
            method: "GET",
            headers: {
                authorization: `Bearer ${staff.token}`
            },
        })

            // Handle the response from backend here
            .then((res) => {
                console.log(res.data);
                let status = "nan";

                if (res.data.Approve && res.data.Complete) {
                    status = "complete";
                }
                else if (res.data.Approve && !(res.data.Complete)) {
                    status = "approve";
                }
                else if (!(res.data.Approve) && !(res.data.Complete) && (res.data.Cancel)) {
                    status = "cancel";
                }
                else if (!(res.data.Approve) && res.data.Pending && !(res.data.Complete)) {
                    status = "pending";
                }
                setConsultation(
                    {
                        ref_num: res.data.ID,
                        student: res.data.Student_email,
                        status: status,
                        note: res.data.Note,
                        cancel_reason: res.data.Cancel_reason,
                        datetime: res.data.DateTime
                    }
                )
            })

            .catch((error) => {
                // alert(error.response.data);
                console.log(error)
            });
    }, []);


    return (
        <Container className="my-5">
            <Breadcrumb>
                <Breadcrumb.Item href="/">Dashboard</Breadcrumb.Item>
                <Breadcrumb.Item href="/staff/appointments">
                    Appointments
                </Breadcrumb.Item>
                <Breadcrumb.Item active>{id}</Breadcrumb.Item>
            </Breadcrumb>
            
            <Button className="my-3" variant="primary" href="/staff/appointments" >Back</Button>

            <Card bg="dark" key="dark" text="white">
                <Card.Header>Ref Num: {consultation.ref_num}</Card.Header>
                <ListGroup variant="flush">
                    <ListGroup.Item><strong>Consultation Date and Time:</strong> {consultation.datetime}</ListGroup.Item>
                    <ListGroup.Item><strong>Student:</strong> {consultation.student}</ListGroup.Item>
                    <ListGroup.Item><strong>Status:</strong> <StatusMsg status={consultation.status} /></ListGroup.Item>
                    {consultation.status == "complete" ?? <ListGroup.Item><strong>Note:</strong> {consultation.note}</ListGroup.Item>}
                    {consultation.status == "cancel" ?? <ListGroup.Item><strong>Cancel reason:</strong> {consultation.cancel_reason}</ListGroup.Item>}
                </ListGroup>
            </Card>
        </Container>
    )
}

export default Appointment